compiler/rustc_privacy/src/lib.rs

```diff
@@ -1862,6 +1878,29 @@ impl SearchInterfaceForPrivateItemsVisitor<'_> {
             }
         }
 
+        let Some(effective_vis) = self.effective_vis else {
+            return false;
+        };
+        let reexported_ev = *effective_vis.at_level(Level::Reexported);
+
+        if !vis.is_at_least(reexported_ev, self.tcx) {
+            if self.in_primary_interface {
```

Comment:
```suggestion
            let lint = if self.in_primary_interface { lint::builtin::PRIVATE_INTERFACE } else { lint::builtin::PRIVATE_BOUNDS };
```
Everything else is common.